### PR TITLE
Add support for Rails 6.1. Drop support for Rails < 6.0

### PR DESCRIPTION
**Pull request by Zolotovmega**
Hello! 

In the PR I add support for Rails 6.1, fix tests and drop support for Rails below 6.0 . Initially I tried to integrate Rails 6.1 in the existing code but could not make test pass. I guess this is due to the fact minitest DSL has been changes. 

Also test throw warnings like 
```
DEPRECATED: global use of must_raise from test/test_helper.rb:39. Use _(obj).must_raise instead. This will fail in Minitest 6.
```

That would be nice to fix in a separate PR.

I believe supporting code for all Rails versions starting 3.2 is not necessarily, that complicates the code and tests. Rails < 6 users can continue using 2.12.1 version.

### CODE REVIEW

**Comment by grosser** [comment]
for the warning: you can `require 'maxitest/global_must'` to shut it up :)

File: .rubocop.yml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 AllCops:
   DisplayCopNames: true
-  TargetRubyVersion: 2.4 # lowest supported version
+  NewCops: disable
```
Comment (grosser) [inline .rubocop.yml:3]: prefer enable and disable the ons that cause trouble explicitly (or leave a TODO for that)

File: .rubocop.yml [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
+  SuggestExtensions: false
```
Comment (grosser) [inline .rubocop.yml:4]:
```suggestion
  SuggestExtensions: false # TODO: enable
```

File: .rubocop.yml [hunk 3]
```diff
@@ -1,6 +1,8 @@
...
+  TargetRubyVersion: 2.5
```
Comment (grosser) [inline .rubocop.yml:5]:
```suggestion
  TargetRubyVersion: 2.5 # lowest supported version
```

File: .travis.yml [hunk 4]
```diff
@@ -20,20 +14,15 @@ env:
...
+    gemfile: gemfiles/rails_6_1.gemfile
   - rvm: 2.6
-    gemfile: gemfiles/rails4.2.gemfile
+    gemfile: gemfiles/rails_6_1.gemfile
   include:
-  - rvm: 2.4 # lowest ruby version
-    env: TASK=rubocop
-  # TODO: fix weird rails testing errors unrelated to the library
-  allow_failures:
   - rvm: 2.5
```
Comment (grosser) [inline .travis.yml:26]:
```suggestion
  - rvm: 2.5 # lowest ruby version
```

File: Appraisals [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
```
Comment (grosser) [inline Appraisals:1]: prefer using wwtd and manually maintained gem files ... keep the previous structure 🤷

Comment (Zolotovmega) [reply Appraisals:1]: Oh cool! I missed it 🙈

**Comment by grosser** [comment]
do you mean minitest-rails dsl changed ?

**Comment by grosser** [comment]
let me get https://github.com/zendesk/stronger_parameters/pull/80 sorted first
wwtd will also go away since it uses GA then :/

**Comment by grosser** [comment]
added 6.1 to https://github.com/zendesk/stronger_parameters/pull/79
once that is merged I'll steal the cleanups from here :)

**Comment by Zolotovmega** [comment]
> do you mean minitest-rails dsl changed ?

Well passing symbol to `get` or `post` would result in an invalid URL because `get` and `post` accept urls not controller actions etc!

**Pull request closed without merging**